apm-agent-plugins/apm-hibernate-search-plugin/apm-hibernate-search-plugin-5_x/src/test/java/co/elastic/apm/agent/hibernate/search/v5_x/HibernateSearch5InstrumentationTest.java

```diff
@@ -0,0 +1,206 @@
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.hibernate.search.v5_x;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+
+import co.elastic.apm.agent.AbstractInstrumentationTest;
+import co.elastic.apm.agent.hibernate.search.DeleteFileVisitor;
+import co.elastic.apm.agent.impl.transaction.TraceContext;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.hibernate.ScrollableResults;
+import org.hibernate.Session;
+import org.hibernate.search.FullTextSession;
+import org.hibernate.search.jpa.FullTextEntityManager;
+import org.hibernate.search.jpa.FullTextQuery;
+import org.hibernate.search.jpa.Search;
+import org.hibernate.search.query.dsl.BooleanJunction;
+import org.hibernate.search.query.dsl.QueryBuilder;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class HibernateSearch5InstrumentationTest extends AbstractInstrumentationTest {
+
+    private Path tempDirectory;
+
+    private EntityManagerFactory entityManagerFactory;
+
+    private EntityManager entityManager;
+
+    @BeforeEach
+    void setUp() throws IOException {
+        tempDirectory = Files.createTempDirectory("HibernateSearch5InstrumentationTest");
+        entityManagerFactory = EntityManagerFactoryHelper.buildEntityManagerFactory(tempDirectory);
+        entityManager = entityManagerFactory.createEntityManager();
+        tracer.startTransaction(TraceContext.asRoot(), null, null).activate();
+    }
+
+    @AfterEach
+    void tearDown() throws IOException {
+        tracer.currentTransaction().deactivate().end();
+        entityManager.close();
+        entityManagerFactory.close();
+        Files.walkFileTree(tempDirectory, new DeleteFileVisitor());
+    }
+
+    @Test
+    public void performLuceneIndexSearch() {
+        saveDogsToIndex();
+
+        FullTextEntityManager fullTextSession = Search.getFullTextEntityManager(entityManager);
+        QueryBuilder builder = fullTextSession.getSearchFactory().buildQueryBuilder().forEntity(Dog.class).get();
+
+        BooleanJunction<BooleanJunction> bool = builder.bool();
+        bool.should(builder.keyword().onField("name").matching("dog1").createQuery());
+
+        Query query = bool.createQuery();
+
+        FullTextQuery ftq = fullTextSession.createFullTextQuery(query, Dog.class);
+
+        List<Dog> result = (List<Dog>) ftq.getResultList();
```

Comment:
Looks like lots of common stuff between tests. Consider adding stuff to `@BeforeEach` and `@BeforeAll` as much as possible